data/plugins/follow/util.rb

```diff
@@ -0,0 +1,139 @@
+  def self.get_minimum_distance(mob, other_mob)
+
+    minimum_distance = 1
+
+    delta_x = other_mob.position.x - mob.position.x
+    delta_y = other_mob.position.y - mob.position.y
+
+    other_mob_size = other_mob.instance_of?(Npc) ? other_mob.definition.size : 1
+
+    if other_mob_size > 1
+      other_mob_size += 1 if other_mob_size == -delta_x + 1 || other_mob_size == -delta_y + 1
+      minimum_distance = other_mob_size if delta_y <= 0 && delta_x <= 0
+    end
+
+    mob_size = mob.instance_of?(Npc) ? mob.definition.size : 1
+
+    if mob_size > 1
+      mob_size += 1 if mob_size == delta_x + 1 || mob_size == delta_y + 1
+      minimum_distance = mob_size if delta_y >= 0 && delta_x >= 0
+    end
+
+    return minimum_distance
+  end
+
+  ##
+  # Generates a path and adds the steps to the given mob's walking queue.
+  #
+  # @param [Mob] mob The following mob.
+  # @param [Mob] other_mob The mob that is being followed.
+  # @param [Integer] minimum_distance The minimum distance that must be kept between the two mobs while following.
+  # @param [Integer] maximum_distance The maximum distance that can be kept between the two mobs while following.
+  # @param [Boolean] simple A flag indicating whether or not the path generated will use a 'simple' pathfinder.
+  def self.walk_to_position(mob, other_mob, minimum_distance, maximum_distance, end_position, simple)
+    position = mob.position
+    distance = position.get_distance(other_mob.position)
+
+    pathfinder = simple ? SIMPLE_PATH_FINDER : FOLLOW_PATH_FINDER
+    path = pathfinder.find(position, end_position)
+
+    path.each do |tile|
+      tile_distance = tile.get_distance(other_mob.position)
+      break if tile_distance < minimum_distance
+
+      mob.walking_queue.add_step(tile)
+
+      break if distance > maximum_distance && tile_distance <= maximum_distance
+    end
+  end
+
+  ##
+  # Resets the following state for the given mob.
+  #
+  # @param [Mob] mob The mob.
+  def self.reset(mob)
+    mob.reset_interacting_mob
+    mob.following = -1
+  end
+
+  ##
+  # Schedules a following task.
+  #
+  # @param [Mob] mob The mob that is doing the following.
+  # @param [Mob] other_mob The mob that is being followed.
+  # @param [Integer] maximum_distance The maximum distance that can be kept between the two mobs while following.
+  # @param [Boolean] simple A flag indicating whether or not this mob will use a 'simple' pathfinder.
+  # @param [Boolean] projectile_clipped A flag indicating whether or not pathfinding should be clipped for projectiles.
+  def self.follow(mob, other_mob, maximum_distance, simple, projectile_clipped)
+
+    mob.following = other_mob.index
+    mob.interacting_mob = other_mob
+
+    schedule 0, true do |task|
+
+      # if we're following a different entity, stop this task
+      # TODO: Could the indices of NPCs and Players collide since they are in different instances of MobRepository..?
+      if mob.following != other_mob.index
```

Comment:
@stevesoltys something similar to combat maybe https://github.com/apollo-rsps/apollo/pull/260/files#diff-70a326256cf9b797a8a087b495c01987R3